x-pack/plugins/apm/public/components/fleet_integration/apm_policy_form/settings_definition/rum_settings.ts

```diff
@@ -4,10 +4,31 @@
  * 2.0; you may not use this file except in compliance with the Elastic License
  * 2.0.
  */
+import * as t from 'io-ts';
+import { either } from 'fp-ts/lib/Either';
 import { i18n } from '@kbn/i18n';
 import { PackagePolicyVars, SettingsRow } from '../typings';
 import { isSettingsFormValid, OPTIONAL_LABEL } from '../settings_form/utils';
 
+const arrayRegex = new RegExp(/\[[^\]]*\]/);
```

Comment:
Do we only want to fail validation for bracket pairs or should we fail validation for any bracket?

For example, with this RegExp, it would not invalidate on values `["foo` or `"foo"]`, even though both are clearly invalid domain origins. Maybe we should just check if any bracket is present (`/[\[\]]/.test(inputAsString)`

Reply:
That's a good point, should we only invalidate it for brackets, shouldn't we check for more invalid characters? If so what exactly characters?!

Reply:
I'm not totally sure how these origins are read. I'm guessing they are rendered through a yaml template and the wrapping brackets mess with the list by adding nested items. But brackets are also invalid characters in a domain name, so it works to just check for brackets anywhere in the input string. something like a hyphen (`-`) could be invalid if it was leading the string (e.g. `- foo`) because that would be equivalent to wrapping in brackets adding a nested list item, but a hyphen within the string could be valid (e.g. `foo-bar.com`).

I think we should just address the brackets for right now and if other characters become problems we can always add to it.

Reply:
I did what you suggested though. https://github.com/elastic/kibana/pull/137228/commits/5efaea4f8e4deaf1e334b7d00d167decdcaa98ba